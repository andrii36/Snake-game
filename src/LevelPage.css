.button-group {
    margin-top: 20px;
    padding: 20px;
 }
.button {
    display: block;
    margin: 10px auto;
    height: 30px;
    width: 120px;
    background-color: rgba(226, 21, 21, 0.8);
    border-radius: 10px;
    font-size: 19px;
    color: white;
    cursor: pointer;
}