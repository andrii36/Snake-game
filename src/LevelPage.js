import React from "react";
import "./LevelPage.css";
import "./App.css"

const LevelPage = () => {
    return (
        <div className="game-area">
            
        </div>
    );
};

export default LevelPage;