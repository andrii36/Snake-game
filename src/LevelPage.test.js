import { fireEvent, render, screen } from "@testing-library/react";
import LevelPage from "./LevelPage";

describe('LevelPage component renders correctly', () => {
    
    it('matches snapshot', () => {
        const {asFragment} = render(<LevelPage/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders all elements', () => {
        render(<LevelPage/>);
        const component = screen.getByTestId('level_page_wrapper');
        expect(component.querySelectorAll('h3')).toHaveLength(1);
        expect(component.querySelectorAll('button')).toHaveLength(3);
    });

    it('click on button works properly', () => {
        const fn = jest.fn();
        render(<LevelPage showGameComponent={fn}/>);
        const component = screen.getByTestId('level_page_wrapper');
        fireEvent.click(component.querySelector('button'));
        expect(fn).toHaveBeenCalled();
    });
});