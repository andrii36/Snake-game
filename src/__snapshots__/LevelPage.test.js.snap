// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LevelPage component renders correctly matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="game-area"
    data-testid="level_page_wrapper"
  >
    <h3>
      Level
    </h3>
    <div
      class="button-group"
    >
      <button
        class="button"
      >
        Easy
      </button>
      <button
        class="button"
      >
        Medium
      </button>
      <button
        class="button"
      >
        Hard
      </button>
    </div>
  </div>
</DocumentFragment>
`;
